src/resolvers/workspace.js

```diff
@@ -0,0 +1,213 @@
+            populate: [
+              {
+                path: 'users',
+                model: 'User'
+              },
+              {
+                path: 'projects',
+                model: 'Project'
+              }
+            ]
+          })).workspaces;
+      } catch (err) {
+        console.error('Error finding workspaces', err);
+        throw new ApolloError('Something went wrong');
+      }
+    },
+
+    /**
+     * Returns workspace info by id
+     * @param {ResolverObj} _obj
+     * @param {ResolverArgs} _args
+     * @param {String} _args.id - workspace id
+     * @param {Context}
+     * @return {Workspace}
+     */
+    workspace: async (_obj, { id }, { user }) => {
+      if (user && !user.id) {
+        throw new ForbiddenError('Only authorized users can do this');
+      }
+
+      let userData;
+
+      // Check user access to the workspace
+      try {
+        userData = await User.findById(user.id);
+      } catch (err) {
+        console.error('Error finding user', err);
+        if (err instanceof MongoError) {
+          throw new ApolloError('Something went wrong');
+        } else {
+          throw new ApolloError('Unknown error');
+        }
+      }
+
+      if (!userData.workspaces.includes(id)) {
+        throw new ForbiddenError('Access denied');
+      }
+
+      // We request the data of the workspace
+      // with expanded users and projects fields
+      try {
+        return await Workspace.findById(id)
+          .populate([
+            {
+              path: 'users',
+              model: 'User'
+            },
+            {
+              path: 'projects',
+              model: 'Project'
+            }
+          ]);
+      } catch (err) {
+        console.error('Error finding workspace', err);
+        if (err instanceof MongoError) {
+          throw new ApolloError('Something went wrong');
+        } else {
+          throw new ApolloError('Unknown error');
+        }
+      }
+    }
+  },
+  Mutation: {
+    /**
+     * Create new workspace
+     * @param {ResolverObj} _obj
+     * @param {ResolverArgs} _args
+     * @param {String} _args.name - workspace name
+     * @param {String} _args.description - workspace description
+     * @param {String} _args.image - workspace image
+     * @param {Context}
+     * @return {String} created workspace id
+     */
+    async createWorkspace(_obj, { name, description, image }, { user }) {
+      if (user && !user.id) {
+        throw new ForbiddenError('Only authorized users can do this');
+      }
+
+      // Perhaps here in the future it is worth passing an array of users
+      let ownerId = user.id;
+
+      try {
+        // Create new workspace in mongo
+        let w = await Workspace.create({
+          name: name,
+          description: description,
+          users: [ ownerId ],
+          image: image
+        });
+
+        // update the list of workspaces at user model
+        await User.findByIdAndUpdate(ownerId, {
```

Comment:
(node:9004) DeprecationWarning: Mongoose: `findOneAndUpdate()` and `findOneAndDelete()` without the `useFindAndModify` option set to false are deprecated. See: https://mongoosejs.com/docs/deprecations.html#-findandmodify-